// @huddle01/server-sdk/dist/auth/index.cjs
'use strict';

var chunkRBTITGYO_cjs = require('../chunk-RBTITGYO.cjs');
require('../chunk-NJUEGIAO.cjs');
require('../chunk-UYXCM27V.cjs');
require('../chunk-CGGVUTEY.cjs');
require('../chunk-OBGZSXTJ.cjs');



Object.defineProperty(exports, "AccessToken", {
  enumerable: true,
  get: function () { return chunkRBTITGYO_cjs.AccessToken; }
});
Object.defineProperty(exports, "DEFAULT_PERMISSIONS", {
  enumerable: true,
  get: function () { return chunkRBTITGYO_cjs.DEFAULT_PERMISSIONS; }
});
Object.defineProperty(exports, "MeetingTokenInputSchema", {
  enumerable: true,
  get: function () { return chunkRBTITGYO_cjs.MeetingTokenInputSchema; }
});
Object.defineProperty(exports, "Permissions", {
  enumerable: true,
  get: function () { return chunkRBTITGYO_cjs.Permissions; }
});
Object.defineProperty(exports, "ROLE_PERMISSIONS", {
  enumerable: true,
  get: function () { return chunkRBTITGYO_cjs.ROLE_PERMISSIONS; }
});
Object.defineProperty(exports, "Role", {
  enumerable: true,
  get: function () { return chunkRBTITGYO_cjs.Role; }
});
Object.defineProperty(exports, "RoleSchema", {
  enumerable: true,
  get: function () { return chunkRBTITGYO_cjs.RoleSchema; }
});
Object.defineProperty(exports, "TokenVerifier", {
  enumerable: true,
  get: function () { return chunkRBTITGYO_cjs.TokenVerifier; }
});
Object.defineProperty(exports, "isDefaultRole", {
  enumerable: true,
  get: function () { return chunkRBTITGYO_cjs.isDefaultRole; }
});
